Makefile

```diff
@@ -1,133 +1,155 @@
 # Read project name from .env file
 $(shell cp -n \.env.default \.env)
+$(shell cp -n \Makefile.local.default \Makefile.local)
 $(shell cp -n \.\/src\/docker\/docker-compose\.override\.yml\.default \.\/src\/docker\/docker-compose\.override\.yml)
 include .env
 
-# Setup PHP Variables based on package version
-PHP_IMAGE=php:$(shell printf '%s' "$(PHP_VERSION)" | sed -e 's/\.//')-fpm
-export PHP_IMAGE
+all: | net build install info
 
-all: | include net build install info
+# Include some usefull goals.
+-include Makefile.tools
+
+# First we need to check if user configured project.
 
 include:
 ifeq ($(strip $(COMPOSE_PROJECT_NAME)),projectname)
 #todo: ask user to make a project name and mv folders.
 $(error Project name can not be default, please edit ".env" and set COMPOSE_PROJECT_NAME variable.)
 endif
 
+
+# Totally clear containers and data inside.
+  RUNNING_CONTAINERS := $(shell docker ps -f name=$(COMPOSE_PROJECT_NAME) --format "{{ .ID }}")
+
+clean: info
+	@echo $(RUNNING_CONTAINERS)
+ifeq ($(strip $(RUNNING_CONTAINERS)),)
+	@echo "Nothing to remove for $(COMPOSE_PROJECT_NAME)"
+else
+	@echo "Removing networks for $(COMPOSE_PROJECT_NAME)"
+	docker-compose down
+endif
+	if [ -d "build" ]; then docker run --rm -v $(shell pwd):/mnt skilldlabs/$(PHP_IMAGE) ash -c "rm -rf /mnt/build"; fi
+
+# Prepare folder structure for project.
 build: clean
-	mkdir -p build
+	mkdir -p build/docroot
+	cp src/composer/docroot/index.php build/docroot
 	mkdir -p /dev/shm/${COMPOSE_PROJECT_NAME}_mysql
 
+# Installing project.
 install:
 	@echo "Updating containers..."
 	docker-compose pull
 	@echo "Build and run containers..."
 	docker-compose up -d
-	docker-compose exec php apk add --no-cache git
-	docker-compose exec -T php composer global require --prefer-dist "hirak/prestissimo:^0.3"
-	make reinstall
-
+	docker-compose exec -T php apk add --no-cache git shadow
```

Comment:
check my last updates on our project, `docker-compose` returns code 2, this can break gitlab-ci script during deployment